### ISSUE TITLE
Improve row selection UI and accessibility

### ISSUE THREAD

**Issue by janvi01** [rowyio/rowy#1507]
**Is your feature request related to a problem? Please describe.**
When we select the bulk row, we cannot distinguish between the selected rows and the unselected ones

When we select bulk rows, clicking on the select all checkbox should deselect the selected rows.

**Describe the solution you'd like**
Improve the UI of the selected rows and improve accessibility.

For example -

how selecting bulk mail looks -

<img width="1412" alt="Screenshot 2023-12-15 at 12 56 44 PM" src="https://github.com/rowyio/rowy/assets/34344234/07a5c7ff-395f-4673-9aa6-74c74020b2d7">

rowy's row selection -

<img width="1768" alt="Screenshot 2023-12-15 at 12 57 44 PM" src="https://github.com/rowyio/rowy/assets/34344234/aa9dc4ad-c71e-4dc8-ba59-ca257ad21449">

**Comment by dhairya-rathod** [comment]
@janvi01 I would like to work on this issue. Are you working on this one already?

**Comment by janvi01** [comment]
Yes @dhairya-rathod I am working on it, you may look for any other issues. Thanks!
